google/cloud/dataproc_spark_connect/session.py

```diff
@@ -408,6 +408,22 @@ def _get_dataproc_config(self):
                 dataproc_config.labels["colab-notebook-kernel-id"] = os.environ[
                     "COLAB_NOTEBOOK_KERNEL_ID"
                 ]
+            default_datasource = os.getenv("DATAPROC_SPARK_CONNECT_DEFAULT_DATASOURCE")
+            if default_datasource:
+                if default_datasource.lower() == "bigquery":
+                    default_bigquery_configs = {
+                        "spark.datasource.bigquery.writeMethod": "direct",
+                        "spark.datasource.bigquery.viewsEnabled": "true",
+                        "spark.sql.legacy.createHiveTableByDefault": "false",
+                        "spark.sql.sources.default": "bigquery",
+                        "spark.sql.catalog.spark_catalog": "com.google.cloud.spark.bigquery.BigLakeMetastoreSparkCatalog"
+                    }
+                    # Merge default configs with existing properties, user configs take precedence
+                    for k, v in default_bigquery_configs.items():
+                        if k not in dataproc_config.runtime_config.properties:
+                            dataproc_config.runtime_config.properties[k] = v
```

Comment:
nit: I think that it can be simplified a little using `setdefault` method:
```python
for k, v in default_bigquery_configs.items():
    dataproc_config.runtime_config.properties.setdefault(k, v)
```